File: Strivers A2Z DSA Sheet/242-ValidAnagram/242-ValidAnagram.cpp
// Last updated: 8/7/2026, 7:13:06 PM
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> mp;
        if(s.size()!=t.size()) return false;
        for(int i=0; i<s.size(); i++){
            mp[s[i]]++;
        }
        for(int i=0; i<t.size(); i++){
            if(mp[t[i]] > 0) mp[t[i]]--;
            else return false;
        }
        return true;
    }
};

File: Strivers A2Z DSA Sheet/242-ValidAnagram/242-ValidAnagram.java
// Last updated: 8/7/2026, 7:13:00 PM
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) return false;
        int[] freq = new int[256];

        for(int i = 0; i<s.length(); i++){
            freq[s.charAt(i)]++;
            freq[t.charAt(i)]--;
        }

        for(int i=0; i<256; i++){
            if(freq[i] != 0) return false;
        }
        return true;
    }
}